pallets/reward-distribution/src/lib.rs

```diff
@@ -193,6 +199,21 @@ pub mod pallet {
 			)?;
 			ext::staking::distribute_reward::<T>(&vault_id, reward, reward_currency_id)?;
 
+			// We check if the amount to transfer is greater than of the existential deposit to
+			// avoid potential losses of reward currency, in case currency is not native
+			let minimum_transfer_amount =
+				<T as orml_tokens::Config>::ExistentialDeposits::get(&reward_currency_id);
+			let expected_rewards =
+				ext::staking::compute_reward::<T>(&vault_id, &caller, reward_currency_id)?;
+
+			if expected_rewards == (BalanceOf::<T>::zero()) {
```

Comment:
Nit: We don't need the brackets do we (?)
```suggestion
			if expected_rewards == BalanceOf::<T>::zero() {
```

Reply:
Probably not :)